src/main/java/org/threeten/extra/YearWeek.java

```diff
@@ -473,6 +473,60 @@ public boolean is53WeekYear() {
     public int lengthOfYear() {
         return (is53WeekYear() ? 371 : 364);
     }
+    
+    //-----------------------------------------------------------------------
+    /**
+     * Returns a copy of this year-week with the specified number of weeks added.
+     * * <p>
+     * This instance is immutable and unaffected by this method call.
+     * 
+     * @param weeksToAdd  the weeks to add, may be negative
+     * @return the year-week with the weeks added, not null
+     */
+    public YearWeek plusWeeks(long weeksToAdd) {
+        if (weeksToAdd == 0) {
+            return this;
+        }
+        if (weeksToAdd < 0) {
+            return minusWeeks(-weeksToAdd);
```

Comment:
...fails because it would overflow.
Although isn't the real range smaller?  `(999,999,999 * 2 + 1) * 52 * (7 /6)` ~ 121,333,333,273 weeks, or about 6 orders of magnitude?